docs/v1/P4Runtime-Spec.mdk

```diff
@@ -3013,6 +3031,13 @@ updates](#sec-batching-and-ordering-of-updates). Members need to be inserted
 first, then the group needs to be created, and finally the match entry can be
 inserted. Therefore, 3 distinct `WriteRequest` batches are required.
 
+It is possible to include several `ActionProfileAction` messages with the same
+exact `action` specification in one `ActionProfileActionSet` message. However,
+the P4Runtime client is encouraged not to do so, as the same can be achieved by
+using the `weight` field. Note that to preserve read-write symmetry, the server
+must not coalesce multiple `ActionProfileAction` messages with the same `action`
+specification into one.
+
```

Comment:
Is there are reason we allow duplicates here but not for the members of action profile groups?  Both have a weight that can be used to "repeat" elements, so I'm not sure why we are making a distinction.  Would it make sense to also disallow duplicates here, as we don't expect anyone to use them, there is a clear alternative (weight), and it simplifies targets to not having to deal with repeats?

Reply:
As far as I'm concerned, the burden on the P4Runtime server would be to have to compare the action specifications to make sure they are all distinct. By allowing repeats, we actually simplify the target IMO, since it can blindly configure the HW (but maybe that depends on the HW).

Note that for traditional action selector programming, we allow 2 distinct members (with distinct member ids) to have the same action specification, which is similar to what is allowed here for one-shot selectors. What we disallow is duplicate member ids in an action profile group definition, which doesn't really have an equivalent for one-shot programming.

Reply:
Makes sense, thanks for the explanation.